### PR TITLE
[#636] move tls to s3

### LINKED ISSUE #631: fix release tagging

### PR DESCRIPTION
**Pull request by alexander-semenets**
this fixes #631 
New approach with TLS certs for the Legion clusters is the next:
- If needed generate the cert with Update_TLS_Certificate jenkins job. This job will get appropriate certificate from S3 and validate if it's OK. If not the new certificate will be signed with Lets Encrypt, encrypted and uploaded to S3.
- Create the cluster as usually. During the cluster creation TLS certificates will be downloaded from S3 and used for cluster setup.
- Once a week Validate_all_certificates job will be being launched at jenkins. The job triggers per cluster Update_TLS_Certificate downstream job which in sequence will check if the Certs at S3 are OK and will regenerate them if not.

REMINDER:
- When the PR is merged all the certificates for the clusters must be updated using Update_TLS_Certificate job
- Validate_all_certificates job should be enabled

### CODE REVIEW

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test app

File: deploy/legionPipeline.groovy [hunk 1]
```diff
@@ -103,6 +103,30 @@ def deployLegion() {
...
+                docker.image("${env.param_docker_repo}/k8s-ansible:${env.param_legion_version}").inside("-e HOME=/opt/legion/deploy -v ${WORKSPACE}/deploy/profiles:/opt/legion/deploy/profiles -u root") {
+                    stage('Reissue TLS Certificates') {
+                        sh """
+                        cd ${ansibleHome} && \
+                        ansible-playbook update-tls-certificate.yml \
+                        ${ansibleVerbose} \
+                        --vault-password-file=${vault} \
+                        --extra-vars "profile=${env.param_profile} \
+                        vault_pass=${vault} \
+                        docker_repo=${env.param_docker_repo} \
```
Comment (kirillmakhonin) [inline deploy/legionPipeline.groovy]: Please remove unused arguments (docker_repo and legion_version)

File: deploy/update-tls-cert.Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                        legion.buildDescription()
+                }
+            }
+        }
+
+        stage('Check and Update TLS Certificates') {
+            steps {
+                script {
+                    legion.ansibleDebugRunCheck(env.param_debug_run)
+                    legion.UpdateTLSCert()
```
Comment (kirillmakhonin) [inline deploy/update-tls-cert.Jenkinsfile]: Please use same function naming notation for all methods

Comment (alexander-semenets) [reply deploy/update-tls-cert.Jenkinsfile]: done

File: deploy/legionPipeline.groovy (hunk 1 above)
Comment (alexander-semenets) [reply deploy/legionPipeline.groovy]: done

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Pull request merged**